src/lib.rs

```diff
@@ -759,6 +759,22 @@ pub trait HasContext {
         offset: i32,
     );
 
+    unsafe fn vertex_attrib_1_f32(&self, index: u32, x: f32);
+
+    unsafe fn vertex_attrib_2_f32(&self, index: u32, x: f32, y: f32);
+
+    unsafe fn vertex_attrib_3_f32(&self, index: u32, x: f32, y: f32, z: f32);
+
+    unsafe fn vertex_attrib_4_f32(&self, index: u32, x: f32, y: f32, z: f32, w: f32);
+
+    unsafe fn vertex_attrib_1_f32_slice(&self, index: u32, v: &[f32; 1]);
```

Comment:
Could we use `&[f32]` for `v` in these? It's usually slightly easier to use and would match `uniform_1_f32_slice`

Reply:
Thanks for the review.
I don't think it's a problem, so I've fixed it.

Reply:
Thank you! Could we also make this change for `vertex_attrib_2_f32_slice`, `vertex_attrib_3_f32_slice`, and `vertex_attrib_4_f32_slice`?

Reply:
Thanks for the review.
Is this okay?

Reply:
Yes this looks great, thank you!